kbase/narrative
# DATAUP-464 bulk import subdirectory

# Description of PR purpose/changes

This fixes a bug where files in subdirectories aren't properly added to the bulk import cell.

Some tests may be lacking, still - this adds a test for importing a file in a subdirectory, but not necessarily via the bulk import cell process. Funny, though, there's no tests that explicitly exercise that code path anyway.

Most of the tests that are there are just shuffled around and encapsulated a bit.

# Jira Ticket / Issue #
https://kbase-jira.atlassian.net/browse/DATAUP-464

# Testing Instructions
* Details for how to test the PR:
* Load a subdirectory of your staging area with a few files. 
* Select them for bulk import and hit the button.
* They should now appear in the file path rows of the configure tab with proper sub paths.
* Once the cell is properly filled out, you can hit the "Show Code" function to see that the files are really in the parameters fields properly.
* Then run the cell and see if they upload!

# Dev Checklist:
